### DISCUSSION TITLE
Cannot run on Ubuntu and Fedora

### DISCUSSION THREAD

**Discussion by nils-bauer** [maxjoehnk/Mizer#753] [Q&A]
Hi there,
I am trying to run Mizer on Linux. I downloaded the .zip from the mizer.live website. After extracting all the files and running the mizer executable, no window appears but the process is running.
When running the software from a terminal I get the following error:

```
MESA: error: ZINK: failed to choose pdev
libEGL warning: egl: failed to create dri2 screen
2024-11-03T14:26:46.383667Z ERROR mizer-tokio-runtime mizer_devices::laser: crates/components/connections/devices/src/laser.rs:33: ether dream discovery failed: Resource temporarily unavailable (os error 11)
```

I have tried this on a freshly installed Fedora 40 and Ubuntu 24.04 VM.

**Comment by maxjoehnk** [comment]
That's odd, I'll try to reproduce this in a vm myself.
Fedora was also in a vm? Could this be a gpu configuration issue in the vm itself?

**Reply by nils-bauer** [reply]
Yes, both Ubuntu 24.04 and Fedora 40 in a VM. I have the same behaviour on my laptop outside of the vm in Fedora 41, but only with this error message:
`2024-11-03T21:11:52.727145Z ERROR mizer-tokio-runtime mizer_devices::laser: crates/components/connections/devices/src/laser.rs:33: ether dream discovery failed: Die Ressource ist zur Zeit nicht verfügbar (os error 11)`

On the same device with Windows 11 (native, not a VM) the program starts without problems.

**Reply by maxjoehnk** [reply]
Ah I might have an idea what the issue is.
There is a long standing issue with the flutter integration I'm using and compositors on linux.
The application is actually running just fine on your system but you don't see anything. On the linux systems I'm using I'm running Mizer without a compositor so it is not an issue for me.
I don't think I'll be able to fix this issue anytime soon as I'm not really sure what's the cause for this. A flutter upgrade introduced this bug a while back but only in combination with nativeshell which is the rust integration I'm using.

The error message regarding the ether dream discovery is unrelated and can be ignored (I should probably label it as a warning instead of an error).

**Comment by maxjoehnk** [comment]
I've fixed this issue in nativeshell and integrated the fix in #763.
It should be published to the website in about 1 hour. 

I've confirmed it working on arch with picom as a compositor as well as pop!os 22.

Please report back if you're facing any issues.

**Reply by maxjoehnk** [reply]
I've just tested it on wayland where I couldn't get the gui to show up. So if you're running wayland that might still not work.
I've tried to run it with xwayland as well but it crashed almost immediately.
As far as I know it should support wayland just fine as the flutter shell for linux is using gtk 3, but I'll have to investigate this further.

**Reply by nils-bauer** [reply]
Great, thanks a lot! 
Now it works under X, but not under wayland as you said. Under X I have problems with the touch input, so it would be great to have wayland support in the future.

**Reply by maxjoehnk** [reply]
Which problems do you have with touch input? Multi touch isn't supported on Linux by flutter currently (as far as I know this is also true for Wayland) but outside of that I had no issues using touch on X in the past.

**Reply by nils-bauer** [reply]
Sorry for the confusion. Touch input is not functioning for me in general under GNOME on X, it's not limited to just Mizer. I have to look into it.

**Comment by shujaatak** [comment]
> Hi there, I am trying to run Mizer on Linux. I downloaded the .zip from the mizer.live website. After extracting all the files and running the mizer executable, no window appears but the process is running. When running the software from a terminal I get the following error:
> 
> ```
> MESA: error: ZINK: failed to choose pdev
> libEGL warning: egl: failed to create dri2 screen
> 2024-11-03T14:26:46.383667Z ERROR mizer-tokio-runtime mizer_devices::laser: crates/components/connections/devices/src/laser.rs:33: ether dream discovery failed: Resource temporarily unavailable (os error 11)
> ```
> 
> I have tried this on a freshly installed Fedora 40 and Ubuntu 24.04 VM.

Same case with Windows! Downloaded zip and try to run mizer executable but no window appears.
